File: Sistemas de ecuaciones no lineales/Punto Fijo/Julia/punto_fijo.jl
using Printf

mutable struct Punto
    x::Float64
    y::Float64
end

function g_1(x::Float64, y::Float64)
    return ((x ^ 2)- (y ^ 2) - x - 3 + (3 * x))/(3)
end

function g_2(x::Float64, y::Float64)
    return -(x + (2 * y) + 1 - (3 * y))/(3)
end

function distancia(p_1::Punto, p_2::Punto)
    return √(((p_1.x - p_2.x) ^ 2) + ((p_1.y - p_2.y) ^ 2))
end

function punto_fijo(x::Float64, y::Float64, tolerancia::Float64, iteraciones::Int64)
    p = Punto(x, y)
    pkm1 = Punto(0.0, 0.0)
    k::UInt32 = 0.0
    e_a::Float64 = 1.0

    @printf("|%-4s|%-24s|%-24s|%-24s|\n", "k", "x^k", "y^k", "E_a")
    #println("|-|-|-|-|")
    
    while true
        pkm1.x = p.x
        pkm1.y = p.y

        @printf("|%-4i|%-24.20LF|%-24.20LF|%-24.20LF|\n", k, p.x, p.y, e_a)

        if k >= iteraciones
            break
        end
        if e_a <= tolerancia
            break
        end

        p.x = g_1(pkm1.x, pkm1.y)
        p.y = g_2(pkm1.x, pkm1.y)

        e_a = distancia(p, pkm1)

        k += 1
    end

    return p
end


function main()
    p = punto_fijo(0.0, 0.0,0.0005, 100)
end

main()

File: Sistemas de ecuaciones no lineales/Punto Fijo/C/puntoFijo.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h> 

typedef struct 
{
	long double x;
	long double y;
}Punto;


long double g_1(long double x, long double y)
{
	long double _x = ((x * x) - y + 0.5)/2;

	return _x;	
}

long double g_2(long double x, long double y)
{	
	long double _y = (-(x * x) - (4 * (y * y)) + (8 * y) + 4)/(8);

	return _y;
}

Punto punto_fijo(long double x_0, long double y_0, long double tolerancia, int iteraciones)
{
	unsigned int k = 0;
	long double x_k, y_k, E_a = 0;
	Punto p, pe;

	p.x = x_0;
	p.y = y_0;
	printf("|%-4s|%-12s|%-12s|%-12s|\n", "k", "x^k", "y^k", "E_a");

	while (1)
	{
		
		pe.x = p.x - x_k;
		pe.y = p.y - y_k;
		
		x_k = p.x;
		y_k = p.y;
		
		//printf("\n%LF, %LF\n", pe.x, pe.y);

	if (pe.x == 0)
	{
		pe.y = pow(pe.y, 2);
		E_a = sqrt(pe.y);
		//printf("1\n");
	}else if (pe.y == 0)
	{
		pe.x = pow(pe.x, 2);
		E_a = sqrt(pe.x);
		//printf("2\n");
	}else
	{
		pe.y = pow(pe.y, 2);
		pe.x = pow(pe.x, 2);
		E_a = sqrt(pe.x + pe.y);
		//printf("3\n");
	}
		
		printf("|%-4d|%-12LF|%-12LF|%-12LF|\n", k, x_k, y_k, E_a);
		
		if(E_a < tolerancia && k > 0) break;
		if(k > iteraciones) break;

		p.x = g_1(x_k, y_k);
		p.y = g_2(x_k, y_k);

		k++;
	}
	
	return p;
}


int main()
{
	Punto p;

	p = punto_fijo(0, 1, 0.005, 10);

	return 0;
}
